assets/semgrep_rules/client/typos.cc

```diff
@@ -0,0 +1,53 @@
+  // ok: brave-raptor-typos
+  if (auto* playlist_service =
+          playlist::PlaylistServiceFactory::GetForBrowserContext(
+              web_contents->GetBrowserContext())) {
+    playlist_service->ConfigureWebPrefsForBackgroundWebContents(web_contents,
+                                                                web_prefs);
+  }
+
+  if (host_content_settings_map) {
+    // ok: brave-raptor-typos
+    if (std::unique_ptr<content::NavigationThrottle>
+            domain_block_navigation_throttle = brave_shields::
+                DomainBlockNavigationThrottle::MaybeCreateThrottleFor(
+                    handle, g_brave_browser_process->ad_block_service(),
+                    g_brave_browser_process->ad_block_service()
+                        ->custom_filters_provider(),
+                    EphemeralStorageServiceFactory::GetForContext(context),
+                    host_content_settings_map,
+                    g_browser_process->GetApplicationLocale())) {
+      throttles.push_back(std::move(domain_block_navigation_throttle));
+    }
+  }
+
+content::StoragePartitionConfig
+BraveContentBrowserClient::GetStoragePartitionConfigForSite(
+    content::BrowserContext* browser_context,
+    const GURL& site) {
+  // ok: brave-raptor-typos
+  if (auto* request_otr_service =
+          request_otr::RequestOTRServiceFactory::GetForBrowserContext(
+              browser_context)) {
+    if (request_otr_service->IsOTR(site)) {
+      CHECK(site.has_host());  // upstream also does this before accessing
+                               // site.host()
+      return content::StoragePartitionConfig::Create(
+          browser_context, site.host(), /*partition_name=*/"request_otr",
+          /*in_memory=*/true);
+    }
+
+  // ruleid: brave-raptor-typos
+  if (request_otr_service =
+          request_otr::RequestOTRServiceFactory::GetForBrowserContext(
+              browser_context)) {
+    if (request_otr_service->IsOTR(site)) {
+      CHECK(site.has_host());
+    }
+
+
+  }
```

Comment:
@thypon I don't understand where this finding is coming from. The message is the brave-raptor-typos message since the one in audit does not have "These types of errors are generally the result of a typo". However, there is no `(see $EXPR1)` at the end of the message. Also, this is a fixed false-positive finding but the self-test works. This looks like an old version of the rule having been applied.

Reply:
Ah it's because we are running the org version of the action as well. Doh! https://github.com/brave/security-action/actions/runs/6489372731/workflow?pr=395

Reply:
> Required workflows cannot be configured to run in the repository the workflow is created in. You should consider creating a separate repository to store your required workflows.

https://docs.github.com/en/actions/using-workflows/required-workflows#restrictions-and-behaviors-for-the-target-repository - I guess that means if it's not for all repos.

> Required workflows can be stored in any repository folder and are not restricted to the .github/workflows folder like normal workflows.

https://docs.github.com/en/actions/using-workflows/required-workflows#restrictions-and-behaviors-for-the-source-repository - Really? Because we were getting devops issues until we moved it...